std/process.d

```diff
@@ -406,9 +414,42 @@ private Pid spawnProcessImpl(in char[][] args,
     auto stdoutFD = getFD(stdout);
     auto stderrFD = getFD(stderr);
 
+    int[2] forkPipe;
+    if (core.sys.posix.unistd.pipe(forkPipe) == 0)
+    {
+        setCLOEXEC(forkPipe[1], true);
+    }
+    else
+    {
+        throw ProcessException.newFromErrno("Could not create pipe to check startup of child");
+    }
+    scope(exit) close(forkPipe[0]);
+
+    static void abortOnError(int forkPipeOut, InternalError errorType, int error) nothrow
+    {
+        core.sys.posix.unistd.write(forkPipeOut, &errorType, errorType.sizeof);
+        core.sys.posix.unistd.write(forkPipeOut, &error, error.sizeof);
+        close(forkPipeOut);
+        core.sys.posix.unistd._exit(1);
+        assert(0);
+    }
+
+    static void ignorePipeErrors() nothrow
+    {
+        import core.sys.posix.signal;
+        import core.stdc.string : memset;
+        sigaction_t ignoreAction;
+        memset(&ignoreAction, 0, sigaction_t.sizeof);
+        ignoreAction.sa_handler = SIG_IGN;
+        sigaction(SIGPIPE, &ignoreAction, null);
```

Comment:
`man sigaction` says:

> During an execve(2), the dispositions of handled signals are reset to the default; **the dispositions of ignored signals are left unchanged**.

It looks like this will cause `SIGPIPE` to be ignored in the child process too, then?